src/App/component/ComparisonPage.jsx

```diff
@@ -0,0 +1,368 @@
+import {useEffect, useState} from 'react'
+import {makeStyles} from '@material-ui/core/styles'
+import ProjectAvatar from './ProjectAvatar'
+import DrawingBoard from './DrawingBoard'
+import Axios from 'axios'
+import moment from 'moment'
+import {Backdrop, CircularProgress, MenuItem, Select} from '@material-ui/core'
+import {connect} from 'react-redux'
+import {Redirect} from 'react-router-dom'
+import { Button } from 'react-bootstrap'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+    minWidth: '30px',
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+  toLeft: {
+    float: 'left',
+  },
+  toRight: {
+    float: 'right',
+  },
+  selectLeft: {
+    width: 300,
+    marginRight: 100,
+    marginBottom: 20,
+  },
+  selectRight: {
+    width: 300,
+    marginLeft: 100,
+    marginBottom: 20,
+  },
+  buttonContainer: {
+    display: 'flex',
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+    minWidth: '30px',
+    alignItems: 'center',
+    width:"67%",
+    justifyContent: "space-between",
+  }
+}))
+
+function ComparisonPage(prop) {
+
+  const classes = useStyles()
+  const [commitListDataLeft, setCommitListDataLeft] = useState([])
+  const [commitListDataRight, setCommitListDataRight] = useState([])
+  const [dataForTeamCommitChart, setDataForTeamCommitChart] = useState({labels: [], data: {}})
+  const [currentProject, setCurrentProject] = useState({})
+
+  const [branchList, setBranchList] = useState([])
+  const [selectedBranchList, setSelectedBranchList] = useState([])
+  const [leftBranchSelected, setLeftBranchSelected] = useState("")
+  const [rightBranchSelected, setRightBranchSelected] = useState("")
+
+  const [open, setOpen] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  useEffect(() => {
+    Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then((response) => {
+        setCurrentProject(response.data)
+      })
+      .catch((e) => {
+        alert(e)
+        console.error(e)
+      })
+  }, [])
```

Comment:
Suggest extract all API calling process into one place, make this file easier to read, and at the same time easily reuse the code that calls our back-end API.

That means, maybe we can create a axios request function like this:

```js
  const sendPVSBackendRequest = async (method, url, jwt) => {
    const baseURL = 'http://localhost:9100/pvs-api';
    const requestConfig = {
      baseURL,
      url,
      method,
      headers: {...(jwt && {'Authorization': jwt})}
    };
    return (await Axios.request(requestConfig))?.data;
  };
```

And create a isolated function that uses `sendPVSBackendRequest`, to call its target API:

```suggestion
  const loadInitialProjectInfo = () => {
    sendPVSBackendRequest('GET', `/pvs-api/project/1/${projectId}`)
      .then((responseData) => {
        if (responseData) {
          setCurrentProject(responseData)
        }
      })
      .catch((e) => {
        alert(e)
        console.error(e)
      })
  }

  useEffect(() => {
    loadInitialProjectInfo() // <<-- Now it has a name, so we can easily know what is its purpose ...
  }, [])
```

The advantage of this is that we can give these actions a name, which is the name of his function, so that we can clearly distinguish what the action is doing and the purpose of execution.
Especially in the callback function of react hooks such as `useEffect`.

So suggest also extract other API-reated blocks into their own function in all places.

```js
const doSomthing = async () => {
  sendPVSBackendRequest('METHOD', `/pvs-api/...`)
      .then((responseData) => {
        if (responseData) {
          ... // DO SOMETHING ....
        }
      })
      .catch((e) => {
        alert(e)
        console.error(e)
      })
}
```